Server/service/networkService.js

```diff
@@ -103,20 +103,25 @@ class NetworkService {
       const response = await ping.promise.probe(job.data.url);
       return response;
     };
-
+  
     let isAlive;
-
+  
     try {
-      const { responseTime, response } =
-        await this.measureResponseTime(operation);
+      const { responseTime, response } = await this.measureResponseTime(operation);
       isAlive = response.alive;
+  
+      // Fetch TTL setting from the monitor
+      const monitor = await this.db.getMonitorById(job.data._id);
+      const ttlDays = monitor.ttlDays || 30; // Default to 30 days if not set
+  
       const checkData = {
         monitorId: job.data._id,
         status: isAlive,
         responseTime,
         message: isAlive
           ? successMessages.PING_SUCCESS
           : errorMessages.PING_CANNOT_RESOLVE,
+        expiry: new Date(Date.now() + ttlDays * 24 * 60 * 60 * 1000) // Set TTL
```

Comment:
We only want to set `expiry` if `monitor.checkTTL` is not undefined right?  Otherwise we don't need to set it as it has a default value

Reply:
Thats right. Let me fix that.